test/parallel/test-pipe-outgoing-message-data-emitted-after-ended.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const common = require('../common');
+const http = require('http');
+const OutgoingMessage = require('_http_outgoing').OutgoingMessage;
+
+const server = http.createServer(common.mustCall(function(req, res) {
+  console.log('Got a request, piping an OutgoingMessage to it.');
+  const outgointMessage = new OutgoingMessage();
+  outgointMessage.pipe(res);
+
+  setTimeout(() => {
+    console.log('Closing response.');
+    res.end();
+    outgointMessage.emit('data', 'some data');
+
+    // If we got here - 'write after end' wasn't raised and the test passed.
+    setTimeout(() => server.close(), 10);
+  }, 10);
```

Comment:
This is going to be _very_ flaky.  Can you refactor the test to not rely on timers?